indico/modules/events/abstracts/controllers/abstract_list.py

```diff
@@ -107,11 +110,34 @@ def _process(self):
 
 
 class RHCreateAbstract(RHAbstractListBase):
-    def _process(self):
+    """Create an abstract from scratch or with existing cloned data"""
+
+    @staticmethod
+    def clone_fields(abstract):
+        field_names = ['title', 'description', 'submission_comment', 'submitted_for_tracks', 'submitted_contrib_type']
+        field_data = {}
+        for f in field_names:
+            field_data[f] = getattr(abstract, f)
+        person_links = []
+        link_attrs = get_simple_column_attrs(AbstractPersonLink)
+        for old_link in abstract.person_links:
+            link = AbstractPersonLink(person=old_link.person)
+            link.populate_from_attrs(old_link, link_attrs)
+            person_links.append(link)
+        field_data['person_links'] = person_links
+        for f in abstract.field_values:
+            field_data[f'custom_{f.contribution_field_id}'] = f.data
```

Comment:
Custom field names are generated dynamically. While this isn't the most robust approach, RFC if it's worth changing.

```python
def make_abstract_form:
...
for custom_field in event.contribution_fields:
   ...
   name = f'custom_{custom_field.id}'
```

Reply:
looks good enough IMHO